src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -28,6 +29,43 @@ public NodeSelectorState(Topology topology)
                 UnlikelyEveryoneFaultedChoiceIndex = 0;
             }
 
+            public NodeSelectorState(Topology topology, NodeSelectorState prevState)
```

Comment:
this should call the other ctor, then you can avoid setting same fields in 2 places in the code, which in a long run might result in desync

Reply:
Done